src/WebServerManager.cpp

```
@@ -2012,10 +2018,25 @@ int WebServerManager::enrichFindSpoolByUid(WiFiClient& client, HTTPClient& http,
-                                           int spoolId, int filamentId, float remainingG, float existingInitialWeight) {
+                                           int spoolId, int filamentId, float remainingG, float existingInitialWeight,
+                                           const char* existingExtraJson, const char* tagFormat) {
     char url[256];
-    StaticJsonDocument<256> patch;
+    StaticJsonDocument<512> patch;
     patch["filament_id"] = filamentId;
+
+    // Only touch extras when writing tag_format. Spoolman PATCH replaces the
+    // entire extra map, so the existing extras (nfc_id, middleware fields like
+    // MMU Gate) must be carried over or they get wiped.
+    StaticJsonDocument<384> existingExtra;
```

Clang (14.0.6)

[warning] variable 'existingExtra' is not initialized

(cppcoreguidelines-init-variables)